### PR TITLE
Add Teams package

### PR DESCRIPTION
**Pull request by EricDahlvang**
Fixes: https://github.com/microsoft/botbuilder-dotnet/issues/5448

### Purpose
Move and rebrand the Teams adaptive package

### Tests
Adding these next

#### General
- [x] I have commented my code, particularly in hard-to-understand areas	
- [x] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: packages/Teams/Microsoft.Bot.Components.Teams.csproj [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <DocumentationFile>bin\$(Configuration)\netstandard2.0\Microsoft.Bot.Builder.Dialogs.Adaptive.xml</DocumentationFile>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <PackageId>Microsoft.Bot.Components.Teams</PackageId>
+    <Description>This library implements .NET support for adaptive dialogs with Microsoft Teams</Description>
+    <Summary>This library implements .NET support for adaptive dialogs with Microsoft Teams</Summary>
+    <ContentTargetFolders>content</ContentTargetFolders>
+    <PackageTags>msbot-component;msbot-action;msbot-trigger</PackageTags>
```
Comment (ryanlengel) [inline packages/Teams/Microsoft.Bot.Components.Teams.csproj:18]: msbot-component tag is included in directory.build.props

**Review by ryanlengel — APPROVED** [review]

**Comment by ryanlengel** [comment]
I've added a yaml file for the Teams package and opened #870 to automate testing, when this is merged in main I can wrap up the pipelines

**Review by peterinnesmsft — APPROVED** [review]
LGTM. Can we also add the project to the Microsoft.Bot.Components.sln while we're at it?

File: packages/Teams/Microsoft.Bot.Components.Teams.csproj (hunk 1 above)
Comment (EricDahlvang) [reply packages/Teams/Microsoft.Bot.Components.Teams.csproj]: thanks for reviewing

removed 'msbot-component'

File: packages/Teams/TeamsComponentRegistration.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
[the remark this answers is not in the document]
Comment (EricDahlvang) [reply packages/Teams/TeamsComponentRegistration.cs]: Removed

thanks for reviewing

File: packages/Teams/TriggerConditions/OnTeamsTabSubmit.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using AdaptiveExpressions;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Conditions;
+using Microsoft.Bot.Connector;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Components.Teams
```
[the remark this answers is not in the document]
Comment (EricDahlvang) [reply packages/Teams/TriggerConditions/OnTeamsTabSubmit.cs]: Fixed! nice catch!

**Pull request merged**